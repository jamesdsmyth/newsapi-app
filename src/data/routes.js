// routes - the categories available from newsapi.org
const routes = [
  {
    route: 'home',
    category: 'general'
  },
  {
    route: 'business',
    category: 'business'
  },
  {
    route: 'entertainment',
    category: 'entertainment'
  },
  {
    route: 'health',
    category: 'health'
  },
  {
    route: 'science',
    category: 'science'
  },
  {
    route: 'sports',
    category: 'sports'
  },
  {
    route: 'technology',
    category: 'technology'
  }
]

export default routes;