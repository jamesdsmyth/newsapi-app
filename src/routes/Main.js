import React, { Component } from 'react';

import { Route, Redirect } from "react-router-dom";

import NewsLanding from '../containers/NewsLanding';
import NewsArticleLanding from '../containers/NewsArticleLanding';

// Main is the container that houses the routes of the app
class Main extends Component {
  render() {
    return (
      <main className="main">
        <Route exact path="/" render={() => (
            <Redirect to="/home"/>
          )}
        />
        {
          this.props.routes.map((item, index) => (
            <Route
              exact
              path={`/${item.route}`}
              key={index}
              render={() => (
                <NewsLanding category={item.category} />
              )}
            />
          ))
        }
        {
          this.props.routes.map((item, index) => (
            <Route
              path={`/${item.route}/:articleId`}
              key={index}
              render={(props) => (
                <NewsArticleLanding {...props} category={item.category} />
              )}
            />
          ))
        }
      </main>
    )
  }
}

export default Main;