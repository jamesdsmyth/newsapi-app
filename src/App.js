import React from 'react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';

import store from './reducers/combinedReducers';

import routes from './data/routes';

import Header from './components/Header';
import Main from './routes/Main';

import './styles/app.scss';

const App = () => (
  <Provider store={store}>
    <Router>
      <div className="page-container">
        <Header routes={routes} />
        <Main routes={routes} />  
      </div>
    </Router>
  </Provider>
)

export default App;