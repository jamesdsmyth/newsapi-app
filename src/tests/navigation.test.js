import React from 'react';
import { mount, configure } from 'enzyme';
import expect from 'expect';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';

import routes from '../data/routes';

configure({ adapter: new Adapter() });

describe('<Navigation /> in the header', () => {
  const wrapper = mount(<App routes={routes} />);
  const index = 0;
  const route = routes[index];

  it('expects .navigation class to be defined', () => {   
    expect(wrapper.find('.navigation')).toBeDefined();
  });

  it('expects .navigation__main class to be defined', () => {   
    expect(wrapper.find('.navigation__main')).toBeDefined();
  });

  it('expects .navigation__main__list-item count to match what was passed in props', () => {   
    expect(wrapper.find('.navigation__main__list-item')).toHaveLength(routes.length);
  });

  it('expects NavLink from react router to be defined', () => {   
    expect(wrapper.find('NavLink')).toBeDefined();
  });

  it('expects NavLink count to match what was passed in props', () => {   
    expect(wrapper.find('NavLink')).toHaveLength(routes.length);
  });

  it('should render the <NavLink /> with the correct href', () => {
    expect(wrapper.find('NavLink').at(index).props().to).toBe(`/${route.route}`);
  });
});