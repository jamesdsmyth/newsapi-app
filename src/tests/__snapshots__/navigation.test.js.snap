// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> in the header renders <Navigation /> and matches the snapshot 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__main"
  >
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/home"
        onClick={[Function]}
      >
        home
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/business"
        onClick={[Function]}
      >
        business
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/entertainment"
        onClick={[Function]}
      >
        entertainment
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/health"
        onClick={[Function]}
      >
        health
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/science"
        onClick={[Function]}
      >
        science
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/sports"
        onClick={[Function]}
      >
        sports
      </a>
    </li>
    <li
      className="navigation__main__list-item"
      onClick={[MockFunction]}
    >
      <a
        aria-current={null}
        className="navigation__main__list-item__link"
        href="/technology"
        onClick={[Function]}
      >
        technology
      </a>
    </li>
  </ul>
</nav>
`;
